utils/react/ReactUtils.js

```diff
@@ -5,6 +5,40 @@
  * Copyright © 2020 Extremely Heavy Industries Inc.
  */
 
+import {isValidElement, createElement, cloneElement} from 'react';
+import {throwIf} from '../js';
+
+/**
+ * Return the display name for either a class-based or functional Component.
+ * @param obj
+ * @return {string}
+ */
 export function getReactElementName(obj) {
-    return obj.type.name || obj.type.displayName;  // Support for class-based and functional cmps, respectively
+    return obj.type.name || obj.type.displayName;
+}
+
+/**
+ * Create a React element from either a HoistComponent or a function returning an element.
+ *
+ * Used by the TabContainer, DashContainer, and DockView APIs to process the 'content' configs
+ * provided to them for their tabs and views.
+ *
+ * @param {(Object|function)} content - HoistComponent or function returning a React element. If a
+ *      function, it may be an 'elemFactory' (as created by elemFactory()), or any function that
+ *      returns a React element. In either case, the function will be called with no arguments.
```

Comment:
Why do we call the function with no args, then clone the element to addProps? Why not call the function with addProps in the first place?

Reply:
The reason for not specifying the “additional” props as arguments to that function, is that the application author of the function should not have to think about those props at all and mixing them in.  The API for the application is to just produce an element.